extension/src/main/kotlin/io/holunda/camunda/bpm/extension/jgiven/SelfUpdatingInstanceSupplier.kt

```diff
@@ -0,0 +1,26 @@
+package io.holunda.camunda.bpm.extension.jgiven
+
+import org.camunda.bpm.engine.RuntimeService
+import org.camunda.bpm.engine.runtime.ProcessInstance
+import java.util.function.Supplier
+
+/**
+ * Implementation holding one process instance id and loading it from runtime service on every request.
+ * This class is intended to be subclassed.
+ */
+open class SelfUpdatingInstanceSupplier(
+  protected val processInstanceId: String,
+  protected val runtimeService: RuntimeService
+) : Supplier<ProcessInstance> {
+
+  /**
+   * Retrieves the process instance.
+   */
+  override fun get(): ProcessInstance {
```

Comment:
As discussed: Might be problematic as we wont get a processInstance once it is finished ... but there _could_ (not sure if and which) be BpmnAware assertions that still do things with the processInstance in Memory.

So a slightly better way would be to keep the last found instance and provide it the ended instance cannot be found anymore.

Reply:
yes